### PR TITLE
Story/ccls 2178 delete app

### PR DESCRIPTION
**Pull request by porritta**

### CODE REVIEW

File: caab-api/open-api-specification.yml [hunk 1]
```diff
@@ -169,6 +169,38 @@ paths:
...
+      operationId: 'removeApplication'
+      parameters:
+        - name: 'id'
+          in: 'path'
+          required: true
+          schema:
+            type: 'integer'
+            format: 'int64'
+            example: '1234567890'
+        - name: 'Caab-User-Login-Id'
```
Comment (PhilDigitalJustice) [inline caab-api/open-api-specification.yml:185]: Is this not a referenced parameter somewhere?

File: caab-service/src/integrationTest/java/uk/gov/laa/ccms/data/api/controller/CaseOutcomeControllerIntegrationTest.java [hunk 2]
```diff
@@ -1,22 +1,19 @@
...
 import uk.gov.laa.ccms.caab.api.CaabApiApplication;
 import uk.gov.laa.ccms.data.api.IntegrationTestInterface;
 
 @SpringBootTest(classes = CaabApiApplication.class)
 @SqlMergeMode(MERGE)
-@Sql(executionPhase = BEFORE_TEST_METHOD, scripts = "/sql/application_tables_drop_schema.sql", config = @SqlConfig(transactionMode = TransactionMode.ISOLATED, errorMode = ErrorMode.CONTINUE_ON_ERROR))
-@Sql(executionPhase = BEFORE_TEST_METHOD, scripts = "/sql/application_tables_create_schema.sql", config = @SqlConfig(transactionMode = TransactionMode.ISOLATED))
-public class CaseOutcomeControllerIntegrationTest extends BaseCaseOutcomeControllerIntegrationTest implements
+@Sql(executionPhase = BEFORE_TEST_METHOD, scripts = "/sql/application_tables_create_schema.sql")
+@Sql(executionPhase = AFTER_TEST_METHOD, scripts = "/sql/application_tables_drop_schema.sql")public class CaseOutcomeControllerIntegrationTest extends BaseCaseOutcomeControllerIntegrationTest implements
```
Comment (PhilDigitalJustice) [inline caab-service/src/integrationTest/java/uk/gov/laa/ccms/data/api/controller/CaseOutcomeControllerIntegrationTest.java:16]: New line?

File: caab-service/src/integrationTest/java/uk/gov/laa/ccms/data/api/service/BaseApplicationServiceIntegrationTest.java [hunk 3]
```diff
@@ -360,6 +360,46 @@ public void testGetApplication_noData() {
...
+  @Sql(scripts = {
+      "/sql/application_insert.sql",
+      "/sql/linked_cases_insert.sql",
+      "/sql/opponent_insert.sql",
+      "/sql/case_outcome_insert.sql",
+      "/sql/prior_authority_insert.sql",
+      "/sql/proceeding_insert.sql",
+      "/sql/scope_limitation_insert.sql"
+  })
+  public void testRemoveApplication_applicationExists_removesSuccessfully() {
```
Comment (PhilDigitalJustice) [inline caab-service/src/integrationTest/java/uk/gov/laa/ccms/data/api/service/BaseApplicationServiceIntegrationTest.java:373]: Do we want to check the existence before and after this test?

**Review by PhilDigitalJustice — CHANGES_REQUESTED** [review]
A couple minor comments

File: caab-api/open-api-specification.yml (hunk 1 above)
Comment (porritta) [reply caab-api/open-api-specification.yml]: It wasn't, but it is now. That and a shared 'id' parameter now added and referenced everywhere.

File: caab-service/src/integrationTest/java/uk/gov/laa/ccms/data/api/controller/CaseOutcomeControllerIntegrationTest.java (hunk 2 above)
Comment (porritta) [reply caab-service/src/integrationTest/java/uk/gov/laa/ccms/data/api/controller/CaseOutcomeControllerIntegrationTest.java]: corrected

File: caab-service/src/integrationTest/java/uk/gov/laa/ccms/data/api/service/BaseApplicationServiceIntegrationTest.java (hunk 3 above)
Comment (porritta) [reply caab-service/src/integrationTest/java/uk/gov/laa/ccms/data/api/service/BaseApplicationServiceIntegrationTest.java:373]: Good spot. That was a bit of a lame test. Corrected.

**Comment by porritta** [comment]
Ready for re-review

**Review by farrell-m — APPROVED** [review]
👍

**Review by PhilDigitalJustice — APPROVED** [review]
LGTM

**Pull request merged**